arch/x86/adler32_avx512.c

```diff
@@ -41,7 +41,14 @@ static inline uint32_t adler32_fold_copy_impl(uint32_t adler, uint8_t *dst, cons
                                           20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37,
                                           38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55,
                                           56, 57, 58, 59, 60, 61, 62, 63, 64);
+#if defined(_MSC_VER)
+# pragma warning( push )
+# pragma warning( disable: 4310 )
+#endif
     const __m512i dot3v = _mm512_set1_epi16(1);
```

Comment:
```c
const __m512i dot3v = _mm512_castps_si512(_mm512_set1_epi16(1));
```
instead?

Reply:
Not really familiar with AVX512 intrinsics, but changed...